### PR TITLE
Increasing test coverage

### PR DESCRIPTION
**Pull request by mmontgomery-square**

### CODE REVIEW

File: pkg/credentials/provider_test.go [hunk 1]
```diff
@@ -51,6 +52,79 @@ func TestNewProviderAssignsSessionDuration(t *testing.T) {
...
+	provider, err := NewProvider(audience, roleArn, &jwtSource, sessionDuration, stsProvider)
+	require.NoError(t, err)
+
+	_, err = provider.Retrieve()
+	require.NoError(t, err)
+
+	require.EqualValues(t,
+		expiration.Round(0).Add(-time.Minute),
+		provider.Expiry.ExpiresAt())
+}
```
Comment (mmontgomery-square) [inline pkg/credentials/provider_test.go:96]: I got a little confused by what was going on here until I read through the AWS source code. The expiration "window" that we supply isn't the raw window value, it's a delta that's used to shorten (not extend) the actual expiration window. So, if we supply a 1-minute expiration and the actual expiration time is 30 minutes, the net expiration time is now 29 minutes.

File: cmd/spiffe-aws-assume-role/cli/mappers/iso_8601_duration_mapper_test.go [hunk 2]
```diff
@@ -10,6 +10,7 @@ import (
 )
 
 func TestStandardLibraryCannotParseIso8601Duration(t *testing.T) {
+	defer func() {}()
 	_, err := time.ParseDuration("PT5M")
 	require.Error(t, err, "Standard library was able to parse ISO8601 duration string, iso8601DurationMapper may no longer be needed")
 }
```
[the remark this answers is not in the document]
Comment (captiosus) [reply cmd/spiffe-aws-assume-role/cli/mappers/iso_8601_duration_mapper_test.go:16]: maybe leave a comment explaining that in the code as well

**Review by captiosus — APPROVED** [review]
LGTM, I would just leave a comment explaining why theres a defer in `TestStandardLibraryCannotParseIso8601Duration`

File: pkg/credentials/provider_test.go (hunk 1 above)
Comment (isemaya-square) [reply pkg/credentials/provider_test.go:96]: Oh weird! So we need a negative window value to expand the actual window.

**Pull request merged**